xds/internal/balancer/clusterresolver/balancerconfig/configbuilder_test.go

```diff
@@ -0,0 +1,867 @@
+				Name: weightedtarget.Name,
+				Config: &weightedtarget.LBConfig{
+					Targets: map[string]weightedtarget.Target{
+						assertString(testLocalityIDs[2].ToString): {
+							Weight: 20,
+							ChildPolicy: &internalserviceconfig.BalancerConfig{
+								Name: lrs.Name,
+								Config: &lrs.LBConfig{
+									ClusterName:             testClusterName,
+									EDSServiceName:          testEDSServiceName,
+									LoadReportingServerName: testLRSServer,
+									Locality:                &testLocalityIDs[2],
+									ChildPolicy:             &internalserviceconfig.BalancerConfig{Name: roundrobin.Name},
+								},
+							},
+						},
+						assertString(testLocalityIDs[3].ToString): {
+							Weight: 80,
+							ChildPolicy: &internalserviceconfig.BalancerConfig{
+								Name: lrs.Name,
+								Config: &lrs.LBConfig{
+									ClusterName:             testClusterName,
+									EDSServiceName:          testEDSServiceName,
+									LoadReportingServerName: testLRSServer,
+									Locality:                &testLocalityIDs[3],
+									ChildPolicy:             &internalserviceconfig.BalancerConfig{Name: roundrobin.Name},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	wantAddrs := []resolver.Address{
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[0][0]}, []string{"priority-2-0", assertString(testLocalityIDs[0].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[0][1]}, []string{"priority-2-0", assertString(testLocalityIDs[0].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[1][0]}, []string{"priority-2-0", assertString(testLocalityIDs[1].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[1][1]}, []string{"priority-2-0", assertString(testLocalityIDs[1].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[2][0]}, []string{"priority-2-1", assertString(testLocalityIDs[2].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[2][1]}, []string{"priority-2-1", assertString(testLocalityIDs[2].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[3][0]}, []string{"priority-2-1", assertString(testLocalityIDs[3].ToString)}),
+		hierarchy.Set(resolver.Address{Addr: testAddressStrs[3][1]}, []string{"priority-2-1", assertString(testLocalityIDs[3].ToString)}),
+	}
+
+	if diff := cmp.Diff(gotNames, wantNames); diff != "" {
+		t.Errorf("buildClusterImplConfigForEDS() diff (-got +want) %v", diff)
+	}
+	if diff := cmp.Diff(gotConfigs, wantConfigs); diff != "" {
+		t.Errorf("buildClusterImplConfigForEDS() diff (-got +want) %v", diff)
+	}
+	if diff := cmp.Diff(gotAddrs, wantAddrs,
+		cmp.AllowUnexported(attributes.Attributes{}),
+		cmp.Transformer("SortAddrs", func(in []resolver.Address) []resolver.Address {
+			out := append([]resolver.Address(nil), in...) // Copy input to avoid mutating it
+			sort.Slice(out, func(i, j int) bool {
+				return out[i].Addr < out[j].Addr
+			})
+			return out
+		}),
```

Comment:
These same `cmpOpts` are used in a bunch of tests. Is it possible to pull them out?

Reply:
Done